cirq-core/cirq/work/sampler.py

```diff
@@ -294,9 +298,26 @@ async def run_batch_async(
         See docs for `cirq.Sampler.run_batch`.
         """
         params_list, repetitions = self._normalize_batch_args(programs, params_list, repetitions)
-        return await duet.pstarmap_async(
-            self.run_sweep_async, zip(programs, params_list, repetitions)
-        )
+        if len(programs) <= self.CHUNK_SIZE:
+            return await duet.pstarmap_async(
+                self.run_sweep_async, zip(programs, params_list, repetitions)
+            )
+
+        results = []
+        for program_chunk, params_chunk, reps_chunk in zip(
+            _chunked(programs, self.CHUNK_SIZE),
+            _chunked(params_list, self.CHUNK_SIZE),
+            _chunked(repetitions, self.CHUNK_SIZE),
+        ):
+            # Run_sweep_async for the current chunk
+            await duet.sleep(1)  # Delay for 1 second between chunk
```

Comment:
Is this delay needed? If we wait for completion of the chunk (as it appears we do), then we should be able to immediately begin another chunk.

Reply:
Yes, without it we still run into the quota exception when polling for results (e.g for 100+ jobs with relatively shallow circuits). 

Empirically, I found that sleeping for 1s removes the error for both shallow and deep circuits